OpenRA.Mods.Common/Traits/Render/WithDecoration.cs

```diff
@@ -69,7 +69,17 @@ public WithDecoration(Actor self, WithDecorationInfo info)
 			Anim.PlayRepeating(info.Sequence);
 		}
 
-		public virtual bool ShouldRender(Actor self) { return true; }
+		protected virtual bool ShouldRender(Actor self)
+		{
+			if (self.World.RenderPlayer != null)
```

Comment:
Well... looking at this again I see you are just moving existing code, so ignore this if you'd like.